firebase-ai/src/main/kotlin/com/google/firebase/ai/type/UsageMetadata.kt

```diff
@@ -30,15 +30,35 @@ import kotlinx.serialization.Serializable
  * candidates.
  * @param thoughtsTokenCount The number of tokens used by the model's internal "thinking" process.
  */
-public class UsageMetadata(
+public class UsageMetadata
+internal constructor(
   public val promptTokenCount: Int,
   public val candidatesTokenCount: Int?,
   public val totalTokenCount: Int,
   public val promptTokensDetails: List<ModalityTokenCount>,
   public val candidatesTokensDetails: List<ModalityTokenCount>,
   public val thoughtsTokenCount: Int,
+  placeholder: Unit
 ) {
 
+  @Deprecated("Not intended for public use")
+  public constructor(
```

Comment:
Is the public constructor used to initialize UsageMetadata anywhere within the code